cmake/modules/CompileSettingsDebug.cmake

```diff
@@ -28,9 +28,14 @@ include(CMakePackageConfigHelpers)
 #
 # Build type specific options
 #
-if("${BUILD_TYPE}" STREQUAL "DebugOptimized")
+if("${BUILD_TYPE}" STREQUAL "Debug")
     if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR ${CMAKE_COMPILER_IS_GNUCXX} )
-        target_compile_options(CompileSettingsDebug INTERFACE -g)
+        target_compile_options(CompileSettingsDebug INTERFACE -O0)
+    endif()
+
+elseif("${BUILD_TYPE}" STREQUAL "DebugOptimized")
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR ${CMAKE_COMPILER_IS_GNUCXX} )
+        target_compile_options(CompileSettingsDebug INTERFACE -O1 -g)
```

Comment:
@bramoosterhuis / @HaseenaSainul just double checking: I would like to have the same optimization level in Release as in Debig so we test the same in both environments. I thought that the Release Optimisation level is 2 (-O2) if so, can we change it to -O2, if it is -O1 this pull request is good to go..

Reply:
I was hoping these flags would come from the build system. If we need to set them our selves I guess we want these flags:
DebugOptimized: "-O2 -g"
Debug: "-O0 -g"

Reply:
Yes,  but usage of -Os flag from build system is creating build issues. So think it is better to define the optimisation level from thunder itself. So update the flags as per your comments